enterprise/tasks.py

```diff
@@ -182,3 +207,133 @@ def send_sso_configured_email(
         )
         LOGGER.exception(message)
         raise exc
+
+
+@shared_task
+@set_code_owner_attribute
+def send_group_membership_invitation_notification(
+    enterprise_customer_uuid,
+    group_membership_uuid,
+    budget_expiration,
+    catalog_uuid
+):
+    """
+    Send braze email notification when member is invited to a group.
+
+    Arguments:
+        * enterprise_customer_uuid (string)
+        * group_membership_uuid (string)
+        * budget_expiration (datetime)
+        * catalog_uuid (string)
+    """
+    enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+    group_membership = enterprise_group_membership_model().objects.filter(
+        uuid=group_membership_uuid
+    ).values().last()
```

Comment:
you can access the related `EnterpriseCustomerUser` record directly from the group membership record, b/c the membership record has a FK to ECU.
Also, if you use `select_related('enterprise_customer_user')`, the related model will be fetched with still only a single query against the DB.

Reply:
...same idea for the Pending ECU record.